x/gamm/keeper/msg_server.go

```diff
@@ -178,7 +179,12 @@ func (server msgServer) SwapExactAmountIn(goCtx context.Context, msg *types.MsgS
 		return nil, err
 	}
 
-	err = server.keeper.chargeTakerFeeSwapAmountIn(ctx, takerFeesCoins, sender, msg.Routes)
+	routesForTakerFee := poolmanagertypes.SwapRoutesUnion{
+		InRoutes:  msg.Routes,
+		OutRoutes: nil,
+		Type:      "in",
+	}
+	err = server.keeper.chargeTakerFee(ctx, takerFeesCoins, sender, routesForTakerFee, msg.Routes[len(msg.Routes)-1].TokenOutDenom)
```

Comment:
isn't it supposed to be msg.Routes[0].TokenOutDenom to get the out token of the first people given that it's an "in" route and this `msg.Routes[len(msg.Routes)-1]` should be for the Out route?